### PR TITLE
HARMONY-1583: Add easy start/stop of services for dev

### PR DESCRIPTION
**Pull request by indiejames**
## Jira Issue ID
HARMONY-1538

## Description
Adds scripts and some cleanup to make it easier to start up the dev environment

## Local Test Steps
1. Check out this branch
2. Stop any running harmony stuff
`./bin/stop-harmony-and-services`
3. Add the following to your .env file
```
CALLBACK_URL_ROOT=http://host.docker.internal:3001
BACKEND_HOST=host.docker.internal
LOCALSTACK_HOST=localhost
LOCAL_DEV=true
```
4. Start the k8s stuff (this will not run harmony, the work scheduler, or the work updaters)
`./bin/bootstrap-harmony`
5. Start the harmony local services  (node)
`./bin/start-dev-services`
6. Test harmony using whatever queries you want
7. Tail the logs  for harmony, work-scheduler, work-updaters in ./logs, ./kubernetes-services/work-scheduler/logs, and ./kubernetes-servicies/work-updater/logs
8. Add a `console.log` or normal `log` statement somewhere in harmony and see harmony restart in the log
9. Stop everything again
`./bin/stop-harmony-and-services`
10. Remove the four lines added in step 3 from .env
11. Run Harmony in a Box
`./bin/bootstrap-harmony`
12. Test harmony using whatever queries you want to verify that HiB works (all pods should be running in the cluster)

## PR Acceptance Checklist
* [x] Acceptance criteria met
* [ ] ~Tests added/updated (if needed) and passing~
* [x] Documentation updated (if needed)

### CODE REVIEW

File: bin/start-all [hunk 1]
```diff
@@ -3,7 +3,10 @@
...
-echo ''
-echo 'Run `bin/reload-services-config` to restart harmony.'
-echo 'Run `kubectl delete namespace harmony` to remove all local resources'
+if [ ! "$LOCAL_DEV" = true ]; then
+  echo 'Running harmony in kubernetes'
+  ./bin/start-harmony
+  echo 'Run `bin/reload-services-config` to restart harmony.'
+fi
+
+echo 'Run ``./bin/stop-harmony-and-services` to remove all local resources'
```
Comment (chris-durbin) [inline bin/start-all:12]: nit extra `

File: package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "coverage": "nyc mocha",
     "start": "NODE_OPTIONS=--max-old-space-size=3096 ts-node -r tsconfig-paths/register --require './app/tracing.ts' app/server.ts",
     "start-dev": "strict-npm-engines && ts-node-dev --no-notify -r tsconfig-paths/register --watch app/views,public/js --respawn app/server",
-    "start-dev-fast": "TS_NODE_TRANSPILE_ONLY=true ts-node-dev --no-notify -r tsconfig-paths/register --respawn --inspect=127.0.0.1:9229 app/server",
+    "start-dev-fast": "TS_NODE_TRANSPILE_ONLY=true ts-node-dev --no-notify -r tsconfig-paths/register --respawn --inspect=127.0.0.1:9200 app/server",
```
Comment (chris-durbin) [inline package.json:13]: may as well add the `${DEBUG_PORT:-9200}` here as well

**Review by chris-durbin — APPROVED** [review]
I tested out the main harmony local development workflow and it looks good. Can someone else test out that harmony in a box is working as expected?

File: bin/start-all (hunk 1 above)
Comment (indiejames) [reply bin/start-all:12]: removed

File: package.json (hunk 2 above)
Comment (indiejames) [reply package.json:13]: done

**Review by ygliuvt — APPROVED** [review]

File: docs/guides/develop.md [hunk 3]
```diff
@@ -137,21 +140,44 @@ minikube ssh grep host.minikube.internal /etc/hosts | cut -f1
...
 
 ```
-$ ./bin/start-postgres-localstack
+./bin/bootstrap-harmony
+./bin/start-dev-services
 ```
 
-This will install Postgres and Localstack and forward their ports to localhost. It will take a few minutes the first time you run it. You will know when it has completed when it prints
+NOTE: You must set `LOCAL_DEV=true` before running these to prevent `bootstrap-harmony` from
+starting harmony and it's support  services in kubernetes.
```
Comment (ygliuvt) [inline docs/guides/develop.md:153]:
```suggestion
starting Harmony and its support services in kubernetes.
```

Comment (indiejames) [reply docs/guides/develop.md]: fixed

**Pull request merged**